### PR TITLE
[INJIMOB-842]: mock the verification of vc only for range error for MOSIP VC.

### PR DESCRIPTION
**Pull request by vijay151096**
## Description
Mock the Verification of VC only for **RangeError** for MOSIP VC and **Sunbird VC**.

## Issue ticket number and link
[INJIMOB-842](https://mosip.atlassian.net/browse/INJIMOB-842)

## References 
Druid : https://druid.obsrv.mosip.net/unified-console.html#workbench
Kafka : https://kafka.obsrv.mosip.net/ui/clusters/main_kafka/topics/mosip-telemetry/messages?q=VC%20verification&filterQueryType=STRING_CONTAINS&attempt=0&limit=100&seekDirection=FORWARD&seekType=OFFSET&seekTo=0::455549

## Screenshots
Problem with druid in consuming the data from kafka to push in DB.
 
This Screen Shot Give us idea of how the data is pushed to kafka, when verification failed 
image

[INJIMOB-842]: https://mosip.atlassian.net/browse/INJIMOB-842?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: shared/vcjs/verifyCredential.ts [hunk 1]
```diff
@@ -70,17 +71,35 @@ export async function verifyCredential(
...
 
   if (!result?.verified) {
-    if (result['results'][0].error.name == 'jsonld.InvalidUrl') {
+    let errorCodeName = result['results'][0].error.name;
+    if (errorCodeName == 'jsonld.InvalidUrl') {
       errorMessage = VerificationErrorType.NETWORK_ERROR;
+      isVerifiedFlag = false;
+    } else if (errorCodeName == VerificationErrorType.RANGE_ERROR) {
+      errorMessage = VerificationErrorType.RANGE_ERROR;
+      const vcIdentifier = verifiableCredential.credentialSubject.UIN
```
Comment (swatigoel) [inline shared/vcjs/verifyCredential.ts:88]: can we add a utility method which returns isUIN -> true or false and simplify the logic here?

File: shared/vcjs/verifyCredential.ts [hunk 2]
```diff
@@ -70,17 +71,35 @@ export async function verifyCredential(
...
+      sendErrorEvent(
+        getErrorEventData(
+          TelemetryConstants.FlowType.vcVerification,
+          TelemetryConstants.ErrorId.vcVerificationFailed,
+          TelemetryConstants.ErrorMessage.vcVerificationFailed + vcIdentifier,
+        ),
+      );
+      isVerifiedFlag = true;
     } else {
       errorMessage = VerificationErrorType.TECHNICAL_ERROR;
```
Comment (swatigoel) [inline shared/vcjs/verifyCredential.ts:100]: we can get rid of else by setting this as default error.
 ```
if (!result?.verified) {
    errorMessage = VerificationErrorType.TECHNICAL_ERROR;
    isVerifiedFlag = false;

    if (errorCodeName == 'jsonld.InvalidUrl') { 
    // update error code 
    } else if (errorCodeName == VerificationErrorType.RANGE_ERROR) {
     // logic for range error 
    }
 }
```

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: shared/vcjs/verifyCredential.ts (hunk 2 above)
Comment (vijay151096) [reply shared/vcjs/verifyCredential.ts]: refactored it.

File: shared/vcjs/verifyCredential.ts (hunk 1 above)
Comment (vijay151096) [reply shared/vcjs/verifyCredential.ts]: Extracted a utility Method.

**Review by swatigoel — APPROVED** [review]

**Pull request merged**